File: app/src/main/java/com/example/maunorafiq/pawangcuaca/presentation/view/LoadDataView.java
package com.example.maunorafiq.pawangcuaca.presentation.view;

import android.content.Context;

/**
 * Created by maunorafiq on 11/29/16.
 */

public interface LoadDataView {

    void showLoading();

    void hideLoading();

    void showRetry();

    void hideRetry();

    void showError(String message);

    Context context();
}

File: app/src/main/java/com/example/maunorafiq/pawangcuaca/presentation/view/LoadDataView.kt
package com.example.maunorafiq.pawangcuaca.presentation.view

import android.content.Context

/**
 * Created by maunorafiq on 11/29/16.
 */

interface LoadDataView {

    fun showLoading()

    fun hideLoading()

    fun showRetry()

    fun hideRetry()

    fun showError(message: String?)

    fun context(): Context
}
